src/client/datascience/editor-integration/codelensprovider.ts

```diff
@@ -161,7 +162,7 @@ export class DataScienceCodeLensProvider implements IDataScienceCodeLensProvider
             return codeWatcher.getCodeLenses();
         }
 
-        // Create a new watcher for this file
+        traceInfoIfCI(`Creating a new watcher for document ${document.uri}`);
         const newCodeWatcher = this.createNewCodeWatcher(document);
```

Comment:
When does this case happen? (the documentManager doesn't contain a document with the matching URI)

Reply:
Document manager doesn't? That seems odd. 

I might have expected the version numbers to not match but the document manager should have all active documents.

Reply:
I'm not saying it does happen, more wondering if this fallback is ever hit.